### PR TITLE
ua: do not re-enable enabled services

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [X] Have you successfully run `make lint`?
- [X] Have you successfully run `pytest tests/unit`?

-----

If a UA service is already enabled, do not attempt to re-enable the service because UA will raise and error.

source: https://bugs.launchpad.net/bugs/1989500
(CRAFT-1450)

### CODE REVIEW

**Comment by mr-cal** [comment]
Spread failure unrelated (https://github.com/snapcore/snapcraft/pull/3958)

File: snapcraft/ua_manager.py [hunk 1]
```diff
@@ -101,8 +128,30 @@ def _is_attached() -> bool:
...
+    listed in the list of service data.
+    """
+    service = next(
+        (service for service in service_data if service.get("name") == service_name),
+        None,
+    )
+
+    if service:
+        # possible statuses are 'enabled', 'disabled', and 'n/a'
+        return service.get("status") == "enabled"
```
Comment (facundobatista) [inline snapcraft/ua_manager.py:147]: This logic feels a little too complext. What about something like the following?
```
    for service in service_data:
        if service.get("name") == service_name):
            # possible statuses are 'enabled', 'disabled', and 'n/a'
            return service.get("status") == "enabled"

    emit.debug(f"Could not find service {service_name!r}.")
    return False
```

**Review by facundobatista — CHANGES_REQUESTED** [review]
Overall it's very good!! I just annotated a small improvement. Thanks!

File: snapcraft/ua_manager.py (hunk 1 above)
Comment (mr-cal) [reply snapcraft/ua_manager.py]: Good idea. Simplified.

**Review by facundobatista — APPROVED** [review]
Awesome, thanks!!

**Comment by mr-cal** [comment]
Rebased onto `hotfix/7.2.2`

**Comment by mr-cal** [comment]
Merging with 2 unrelated failing tests (resolved by https://github.com/snapcore/snapcraft/pull/3953 and https://github.com/snapcore/snapcraft/pull/3962)

**Pull request merged**